plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -165,6 +214,75 @@ void startUpdateWorker() {
         }
     }
 
+    boolean existPipeline(String pipelineId) {
+        GetRequest request = new GetRequest(PipelineStore.INDEX, PipelineStore.TYPE, pipelineId);
+        GetResponse response = client.get(request).actionGet();
+        return response.isExists();
+    }
+
+    Iterable<SearchHit> readAllPipelines() {
+        // TODO: the search should be replaced with an ingest API when it is available
+        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
+        sourceBuilder.version(true);
+        sourceBuilder.sort("_doc", SortOrder.ASC);
+        SearchRequest searchRequest = new SearchRequest(PipelineStore.INDEX);
+        searchRequest.source(sourceBuilder);
+        searchRequest.scroll(scrollTimeout);
+        searchRequest.indicesOptions(IndicesOptions.lenientExpandOpen());
+        SearchResponse searchResponse = client.search(searchRequest).actionGet(scrollTimeout);
+
+        if (searchResponse.getHits().getTotalHits() == 0) {
+            return Collections.emptyList();
+        }
+        logger.debug("reading [{}] pipeline documents", searchResponse.getHits().totalHits());
+        return () -> new SearchScrollIterator(searchResponse);
+    }
+
+    class SearchScrollIterator implements Iterator<SearchHit> {
```

Comment:
would it make sense to add this nice iterator to core? the ingest plugin could reuse it in this case.